interpreters/wamr/Toolchain.defs

```diff
@@ -119,9 +52,16 @@ else ifeq ($(CONFIG_ARCH_SIM),y)
   else
     WTARGET = x86_64
   endif
+else
+
+  # Flags for other architectures (arm or riscv)
+
+  WTARGET = $(LLVM_ARCHTYPE)
+  WCPU    = $(LLVM_CPUTYPE)
+
 endif
 
-RCFLAGS += --target=$(WTARGET) $(WCPU)
+RCFLAGS += --target=$(WTARGET) --target-abi=$(LLVM_ABITYPE) --cpu=$(WCPU)
```

Comment:
should we move this file to nuttx/tools/ folder like rust and zig?

Reply:
I think it's a bit different with Rust/Zig,since them can be used with kernel (link into nuttx),but wmar aot can't, it's highly depends on application side's wamr virtual machine.

So it's better to keep it in apps side.

Reply:
But many activities are bringing WebAssembly into Linux kernel e.g.:
https://github.com/wasmerio/kernel-wasm
https://github.com/eunomia-bpf/wasm-bpf
BTW, it's wrong to couple Makefile with a particular engine(wamr here).

Reply:
Compile .wasm from souce code can be placed in common place since it's engine independent.

But for aot, it's engine dpendent.

Reply:
Yes, please split the compile/link from aot.

Reply:
OK, wasm compilation and aot compilation splitted now.